Wikipedia/Code/WMFAppViewController.m

```diff
@@ -425,64 +420,75 @@ - (void)resumeApp {
     if (![self uiIsLoaded]) {
         return;
     }
-
-    [self.statsFunnel logAppNumberOfDaysSinceInstall];
-
-    [[WMFAuthenticationManager sharedInstance] loginWithSavedCredentialsWithSuccess:^(WMFAccountLoginResult * _Nonnull success) {
-        DDLogDebug(@"\n\nSuccessfully logged in with saved credentials for user '%@'.\n\n", success.username);
-    }
-                                                         userAlreadyLoggedInHandler:^(WMFCurrentlyLoggedInUser * _Nonnull currentLoggedInHandler) {
-                                                             DDLogDebug(@"\n\nUser '%@' is already logged in.\n\n", currentLoggedInHandler.name);
-                                                         }
-                                                                            failure:^(NSError * _Nonnull error) {
-                                                                                DDLogDebug(@"\n\nloginWithSavedCredentials failed with error '%@'.\n\n", error);
-                                                                            }];
-
-    [self startContentSources];
-
-    NSUserDefaults *defaults = [NSUserDefaults wmf_userDefaults];
-    NSDate *feedRefreshDate = [defaults wmf_feedRefreshDate];
-    NSDate *now = [NSDate date];
-
-    BOOL locationAuthorized = [WMFLocationManager isAuthorized];
     
-    if (!feedRefreshDate || [now timeIntervalSinceDate:feedRefreshDate] > WMFTimeBeforeRefreshingExploreFeed || [[NSCalendar wmf_gregorianCalendar] wmf_daysFromDate:feedRefreshDate toDate:now] > 0) {
-        [self updateFeedSourcesWithCompletion:NULL];
-    } else if (locationAuthorized != [defaults wmf_locationAuthorized]) {
-        [self.exploreViewController updateNearby:NULL];
-    }
+    dispatch_block_t done = ^{
```

Comment:
Did you consider pulling all the code in the `done` block into a separate method? Honest question. I'm actually not sure which is more clear and readable.